packages/gitbook/src/app/(site)/(content)/[[...pathname]]/page.tsx

```diff
@@ -165,7 +165,7 @@ export async function generateMetadata(props: PageProps): Promise<Metadata> {
         title: getTitle(data),
         description: page.description ?? '',
         alternates: {
-            canonical: absoluteHref(getPagePath(pages, page), true),
+            canonical: absoluteHref(getPagePath(pages, page), true).replace(/\/+$/g, ''),
```

Comment:
In fact I think the fix should be in `absoluteHref`, but it's just the case where `getPagePath` returns an empty string, then we can append it to `{host}/`.

Reply:
Yeah, sorry I didn't follow the whole thing. But by analyzing deeply it only happens on the root URL because `basePath()` and `baseUrl()` ends with a "/". So I fixed it here as you suggested.